src/main.zig

```diff
@@ -402,6 +421,36 @@ fn isValidIdentifierName(str: []const u8) bool { // TODO: Remove after #480
 	}
 	return true;
 }
+const StartupStatus = enum {
+	showController,
+	showNamePrompt,
+	showMainWindow,
+	finished
+};
+var startup_status: StartupStatus = .showController;
+pub fn openNextStartupWindow() void {
+	switch(startup_status) {
+		.showController => {
+			startup_status = .showNamePrompt;
+			gui.openWindow("download_controller_mappings");
+			openNextStartupWindow();
+		},
+		.showNamePrompt => {
+			startup_status = .showMainWindow;
+			if (settings.playerName.len == 0) {
+				gui.openWindow("change_name");
+			} else {
+				openNextStartupWindow();
+			}
+		},
+		.showMainWindow => {
+			startup_status = .finished;
+			gui.openWindow("main");
+		},
+		.finished => { }
+	}
+}
```

Comment:
Do we even need this now? If you open both windows anyways, then I'd prefer to just go back to the previous solution which didn't require this extra function.

Reply:
I undid that and merged.